### PR TITLE
Glint setup

### PR DESCRIPTION
**Pull request by natmegs**
### :pushpin: Summary

This PR sets up some foundations for converting packages/components to TS/glint. These changes were validated by enabling glint type checking for the `Hds::Interactive` and `Hds::Button` component (previously converted to TS). The main changes in this PR:
* Adds a couple glint packages and configs
* Adds a glint linting step 
* Adds nocheck comments to all untyped components (this allows us to incrementally convert components, nocheck comment to be removed once converted)
* Updates ember-truth-helpers to a version that exposes a glint type registry

***

### 👀 Component checklist

- [x] Percy was checked for any visual regression
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Comment by aklkv** [comment]
would it make sense to follow the same path as I did with [core](https://github.com/hashicorp/cloud-ui/pull/9048)? I feel like this PR doesn't immediately address consumer typings but if we follow the above steps it would add types as a first step. there will need to be some adjustments to rollup config, examples of which can be found [here](https://github.com/hashicorp/cloud-ui/pull/9175). Happy to chat more about this approach in slack

**Review by alex-ju — APPROVED** [review]
This approach makes sense to me for a gradual migration! 👏 

I would be keen to hear more from Alexey about what he considers a difference in approaches. Is it about the nested colocated components or is it about something else?

File: packages/components/package.json [hunk 1]
```diff
@@ -48,7 +48,7 @@
     "ember-modifier": "^4.1.0",
     "ember-stargate": "^0.4.3",
     "ember-style-modifier": "^3.0.1",
-    "ember-truth-helpers": "^3.1.1",
+    "ember-truth-helpers": "^4.0.3",
```
[the remark this answers is not in the document]
Comment (WenInCode) [reply packages/components/package.json:51]: @alex-ju would this PR require 2 changesets? or do you group the general purpose of this PR,  the enabling glint, and the ember truth helpers upgrade under one changeset with two entries. I think 2 changesets makes sense given they get transformed properly in the release, just figured I'd confirm.

[the remark this answers is not in the document]
Comment (alex-ju) [reply packages/components/package.json:51]: having two entries sounds good to me

**Comment by natmegs** [comment]
@aklkv just to clarify, is your suggestion to switch to colocated components?

**Comment by aklkv** [comment]
I think co-located component structure will be helpful when the time comes to move to gts but not required currently and consumers are not affected. what I am mainly talking about is to provide component signatures to consumers which directly impacts consumers by first converting all components backing classes to ts and then running codemod, glint adoption at this point is not required but nice to have. Also a side note here is another helpful script to generate a template registry for consumers which is only required when using glint in loose mode https://github.com/bertdeblock/generate-template-registry

**Comment by natmegs** [comment]
@aklkv I think we're on the same page - after this PR, we plan to convert the components to TS one by one. In this process, each component will get component signature types which will be added to the existing [template-registry](https://github.com/hashicorp/design-system/blob/main/packages/components/src/template-registry.ts). Is this in line with what you're thinking?

**Review by WenInCode — APPROVED** [review]

**Pull request merged**